### DISCUSSION TITLE
SQL Injection Not Blocked by ORM

### DISCUSSION THREAD

**Discussion by atspear** [wheels-dev/wheels#1266] [General]
I just had a security scan done on my application and was flagged for SQL Injection vulnerability.  I assumed it was a false positive since I'm using CFWheels ORM, but I tried it and it does indeed work.  The SQL basically sends a sleep command to MySQL and causes the page response to return much slower...

*** Code Snippet Removed ***

My code is pretty simple...

*** Code Snippet Removed ***

I'm running CFWheels 2.3.0 and Lucee 5.3.10

**Comment by bpamiri** [comment]
@atspear I've removed the code snippets till the issue is investigated and a resolution can be implemented.

**Comment by adamcameron** [comment]
To be completely honest I would not rely on the CFWheels value-extraction code to perform anything more than superficial SQLi protection. It's pretty easy to defeat, and the implementation is pretty impenetrable (largely one "clever" regex pattern...) so difficult to know what vectors it doesn't deal with.

Clean your values "by hand" in your app before passing them to CFWheels.

**Comment by chapmandu** [comment]
I recall the (undocumented) 'parameterize=x' argument for find* methods to only allow x parameters to be processed. In your case 'parameterize=1'.
